maven-settings-builder/src/main/java/org/apache/maven/settings/building/DefaultSettingsBuilder.java

```diff
@@ -241,20 +240,17 @@ private Settings interpolate( Settings settings, SettingsBuildingRequest request
                 + e.getMessage(), -1, -1, e );
         }
 
-        interpolator.addPostProcessor( new InterpolationPostProcessor()
+        interpolator.addPostProcessor( ( expression, value ) ->
         {
-            @Override
-            public Object execute( String expression, Object value )
+            if ( value != null )
             {
-                if ( value != null )
-                {
-                    // we're going to parse this back in as XML so we need to escape XML markup
-                    value = value.toString().replace( "&", "&amp;" ).replace( "<", "&lt;" ).replace( ">", "&gt;" );
-                    return value;
-                }
-                return null;
+                // we're going to parse this back in as XML so we need to escape XML markup
+                value = value.toString().replace( "&", "&amp;" ).replace( "<", "&lt;" ).replace( ">", "&gt;" );
```

Comment:
not changed in this PR, but dollars to donuts this conversion is buggy

Reply:
I had to google `dollars to donuts` to understand your comment :)

Do you mean not only these conversions should be done? https://stackoverflow.com/questions/1091945/what-characters-do-i-need-to-escape-in-xml-documents

Reply:
Nothing specific and not this PR's problem to fix. It just happened to call my attention to it. I've just learned from experience that these sorts of replacements always go wrong. Just a note to myself to dig into this, really.